pyomo/contrib/mindtpy/tests/test_mindtpy_global.py

```diff
@@ -34,69 +33,71 @@
-                                mip_solver=required_solvers[1],
-                                nlp_solver=required_solvers[0],
-                                add_nogood_cuts=True,
-                                bound_tolerance=1E-5)
-
-            self.assertIn(results.solver.termination_condition, [
-                          TerminationCondition.optimal, TerminationCondition.feasible])
-            self.assertAlmostEqual(value(model.cost.expr), 68, places=1)
-
-    def test_GOA_8PP_init_max_binary(self):
-        """Test the outer approximation decomposition algorithm."""
-        with SolverFactory('mindtpy') as opt:
-            model = EightProcessFlowsheet()
-            print('\n Solving 8PP problem with Outer Approximation(max_binary)')
-            results = opt.solve(model, strategy='GOA',
-                                init_strategy='max_binary',
-                                mip_solver=required_solvers[1],
-                                nlp_solver=required_solvers[0],
-                                add_nogood_cuts=True)
-
-            self.assertIn(results.solver.termination_condition, [
-                          TerminationCondition.optimal, TerminationCondition.feasible])
-            self.assertAlmostEqual(value(model.cost.expr), 68, places=1)
-
-    def test_GOA_8PP_L2_norm(self):
-        """Test the global outer approximation decomposition algorithm."""
-        with SolverFactory('mindtpy') as opt:
-            model = EightProcessFlowsheet()
-            print('\n Solving 8PP problem with Outer Approximation(L2_norm)')
-            results = opt.solve(model, strategy='GOA',
-                                mip_solver=required_solvers[1],
-                                nlp_solver=required_solvers[0],
-                                feasibility_norm='L2',
-                                add_nogood_cuts=True)
-
-            self.assertIn(results.solver.termination_condition, [
-                          TerminationCondition.optimal, TerminationCondition.feasible])
-            self.assertAlmostEqual(value(model.cost.expr), 68, places=1)
-
-    def test_GOA_8PP_sympy(self):
-        """Test the global outer approximation decomposition algorithm."""
-        with SolverFactory('mindtpy') as opt:
-            model = EightProcessFlowsheet()
-            print('\n Solving 8PP problem with Outer Approximation(sympy)')
-            results = opt.solve(model, strategy='GOA',
-                                mip_solver=required_solvers[1],
-                                nlp_solver=required_solvers[0],
-                                differentiate_mode='sympy',
-                                add_nogood_cuts=True)
-
-            self.assertIn(results.solver.termination_condition, [
-                          TerminationCondition.optimal, TerminationCondition.feasible])
-            self.assertAlmostEqual(value(model.cost.expr), 68, places=1)
+    # def test_GOA_8PP(self):
```

Comment:
Instead of removing these tests entirely, what about leaving them and just skipping them if baron is not licensed (like in `pyomo/contrib/gdpopt/tests/test_LBB.py`)?  Is this because you are using Baron through GAMS and not directly?

Reply:
This is actually an issue with BARON not having a license when it is installed in the testing computers. That particular example:
a) Could not be solved to global optimality using IPOPT, so the assert against the global optimal solution reported by Kesavan et al. would fail.
b) The problem has 11 total variables, and BARON without a license has a hard limit of 10 variables max for it to work.
We decided to remove the test because it wasn't testing any particular part of the code that the other tests were testing (regression testing issues) and because we didn't know how to verify if the computer where the test was executed had a valid license for BARON.

Reply:
Right.  We have a Baron license on the Jenkins testing server at SNL, but only for the direct Baron interface.  The tests on GHA and Travis only use unlicensed (demo mode) solvers.  If we leave the tests in and just guard them with checks for if the Baron license is available, then we would be OK.